misc/python/materialize/output_consistency/generators/expression_generator.py

```diff
@@ -138,25 +160,77 @@ def _generate_args_for_operation(
 
         args = []
 
-        for arg_index in range(0, number_of_args):
+        must_use_aggregation = False
+        for arg_index in range(FIRST_ARG_INDEX, number_of_args):
             param = operation.params[arg_index]
-            arg = self.generate_arg_for_operation_param(param, storage_layout)
+            # nesting_level was already incremented before invoking this function
+            arg = self._generate_arg_for_param(
+                operation,
+                param,
+                arg_index,
+                storage_layout,
+                must_use_aggregation,
+                nesting_level,
+            )
             args.append(arg)
 
+            if arg.is_aggregate:
+                # first argument is aggregated, therefore all must be aggregated (or constant, currently not supported)
```

Comment:
What do you mean by a random element? A constant expression? I could but I think favoring further aggregate expressions reveal more bugs than just constants.

Reply:
The MySQL/MariaDB thing is a historical bad decision and one of the many reasons MySQL had a Mongo-like reputation back in the day.  It has since been remedied - https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html . It is the only known database that has/had this hole.

@def- what version are you running? It may be time to upgrade :-)

Reply:
Heh, good to know. MariaDB 10.5 still does this.

Reply:
yes apparently the default in MariaDB is the unsafe one :-( https://mariadb.com/kb/en/sql-mode/